### PR TITLE
Export tiles and tilesets

### PR DESCRIPTION
**Pull request by gabre**

### CODE REVIEW

File: common/tmx_level.go [hunk 1]
```diff
@@ -72,24 +72,24 @@ type TMXData struct {
 	// valid values are one of "", "zlib", "gzip"
 	Compression string `xml:"compression,attr"`
 	// Data contains an encoded list of uint32 guids defining
-	// the tile layout of a layer
+	// the Tile layout of a layer
 	Data string `xml:",innerxml,"`
 	// Tiles is an array of tiles containing guids. Not set if
 	// other encodings are used
-	Tiles []TMXTile `xml:tile`
```
Comment (Noofbiz) [inline common/tmx_level.go]: Don't change the xml parsing comment to be capitalized, that may mess up unmarshalling

Comment (gabre) [reply common/tmx_level.go]: Ah that was Goland's super renaming refactoring functionality. Sorry, one has to do everything manually.

Comment (Noofbiz) [reply common/tmx_level.go]: No worries, I only knew to be wary of that because I've done exactly that myself before lol

**Comment by EtienneBruines** [comment]
We may want to reference that this fixes #484

**Comment by gabre** [comment]
I ran a manual test:
- first, I ran TMX demo from this PR, it worked fine (note that I changed TMX demo's map file)
- then, I cherry-picked the modified map file onto master where it crashed (panic: runtime error: index out of range)

**Comment by Noofbiz** [comment]
Looks good. Just ran all the demos that use tile maps in some capacity, and all of them are the same, as expected. Thank you for your contribution!

**Pull request merged**